### DISCUSSION TITLE
any integration with headless CMS like strapi?

### DISCUSSION THREAD

**Discussion by ctrngk** [blitz-js/blitz#1503] [Q&A]
Hi all,
For people who don't know, strapi is a headless CMS, which can create a restful API within a few minutes. When I looked back other auth strategy, like pasport.js, I do not want to go back any more. It has strong convention over configuration, generating a standard restful API with backend files by clicking instead of programming. 

My question is, how to integrate with blitz?  If I use blitz, does it mean I have to give up such brilliant CMS? 

Regards,

**Comment by rizrmd** [comment]
Blitzjs is using nextjs under the hood, so I think you can use nextjs integration in this docs: https://nextjs.org/docs/advanced-features/preview-mode

**Comment by flybayer** [comment]
Yeah, you can definitely use Sanity! Best thing for now is to follow Sanity docs for using with Next.js. The integration will be the same for Blitz

**Reply by danielemonaca** [reply]
Just Sanity? Can't you use any headless CMS that is usable in Next.js?

**Reply by flybayer** [reply]
Oops, I mean "stapi" lol. Yes, you can use any headless CMS!

**Comment by mehdiminaee** [comment]
After using `blitz g all example a:string b:sring? c:boolean:true` code, the page layer just create one visually field, how can i generate all visually fields in my project?
Regards, Mehdi Minaee

**Reply by beerose** [reply]
Blitz currently doesn't generate all the fields — it only generates `name` field in form components. You'll have to add the remaining ones manually.
